### DISCUSSION TITLE
Content Editor Components - Slider

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2896] [General] [answered]
Hey team! Sliders are alllllll over the place in the Image Editor I'm about to start Pastify-ing as part of the Content Editor work, and I wanted to add some additional weight behind [this request](https://github.com/twilio-labs/paste/discussions/1627) and learn more about how I can help, and if there's anything in the works I can build from.

![Screenshot 2022-12-15 at 1 23 19 PM](https://user-images.githubusercontent.com/87141551/207959419-b192af0f-e684-4a40-81ec-92cc6d4fc2f7.png)

Thank you!

cc: @brannytempy

**Comment by kelseycavitt** [comment, marked as answer]
@ceyerkes So, the slider is in our backlog, but not on our roadmap for Q1 at this time. 

Our stance so far, until we have a Paste component in place, has been “do what you used to do”. From the looks of the design, you may be able to use text inputs to allow the user to input information, especially if it’s an MVP. Another option is to style a native [<input type=range>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range) HTML element to look like Paste in code. If you want to use a well built primitive, check out how [Adobe Spectrum](https://react-spectrum.adobe.com/react-aria/useSlider.html) does this for their slider component, then also style it to look like Paste!

Let us know if you more questions or what you decide to do, best of luck!

**Reply by ceyerkes** [reply]
Thanks @kelseycavitt! I'm going to use this little -/+ input for now, but happy to swap it out when something better becomes available.

![Screenshot 2023-01-19 at 9 11 04 AM](https://user-images.githubusercontent.com/87141551/213493770-4954fcff-d033-4b33-949f-33ab8c1a520e.png)

**Reply by ceyerkes** [reply]
After chatting it over with my PM, I'm hoping to move towards something like this input:

<img width="278" alt="Screenshot 2023-01-25 at 5 13 34 PM" src="https://user-images.githubusercontent.com/87141551/214723196-85f01114-f949-438d-8b1a-2d7c4b5d1a31.png">

It feels cleaner and more contained, but would love some async feedback whenever makes sense. Thanks!

**Comment by libertycarlin** [comment]
I agree with you @ceyerkes I prefer the second option! The up and down arrows feel like they hold the same significance as the percentage amount, and that's not the case. 

I like the second option because it simplifies the use of the shapes and something as simple as a percentage field isn't very long, so could get away with putting the arrows in the same field container.

**Reply by ceyerkes** [reply]
Sweet, thanks @libertycarlin! I'll consider this one gtg.

**Reply by gloriliale** [reply]
@ceyerkes one more thing: we don't currently have an input that looks like this with the arrows going both ways. Our [select](https://paste.twilio.design/components/select) has only the down chevron but it's the closest, though it would also only allow discreet zoom options.

**Reply by ceyerkes** [reply]
@gloriliale sounds good. Is it ok if I use what I have above for the editor until something similar is available in paste?

**Reply by gloriliale** [reply]
I don't think we have anything like the two arrow input in our backlog, but please correct me if I'm wrong on that @serifluous. We can add it to our backlog if you do a [dependency request](https://airtable.com/shrvMvL3BG6Zm0f9l) though!

**Reply by serifluous** [reply]
The closest thing we have is "Number input with stylized buttons" (which our team may just want to genericize to "Stylized number input"). If you think this UI would still work better with a slider overall, though, we still have that request from you, and there's no need to file another request.

And for now, yep, using this design is likely your best path forward because it can be built with components out-of-the-box. 
![image](https://user-images.githubusercontent.com/3821840/215227951-862a9c53-f47b-464a-8c30-62caed5120db.png)

**Reply by ceyerkes** [reply]
Hey @serifluous - would it be possible for us to use the new design with up/down chevrons instead? We have many instances of inputs needing to sit next to each other, and the 3 buttons in a row design isn't ideal in those scenarios.

**Reply by ceyerkes** [reply]
This input is in almost every module:

<img width="616" alt="Screenshot 2023-01-30 at 8 49 53 AM" src="https://user-images.githubusercontent.com/87141551/215525345-7555692f-b640-4e75-ac14-d49b371301c2.png">

And many modules would gain valuable vertical space by placing 2 inputs side by side:

<img width="131" alt="Screenshot 2023-01-30 at 8 51 21 AM" src="https://user-images.githubusercontent.com/87141551/215525715-fc6d98b2-7089-4de4-b2ea-7da7d5ad7150.png">

**Reply by serifluous** [reply]
What's the interaction on the inputs? Does clicking the up or down arrow change the input value?

**Reply by ceyerkes** [reply]
Yes - clicking up or down changes the input value.

**Reply by ceyerkes** [reply]
Sorry - i think i'm confusing things by conflating the slider and the other number inputs. The 3 button +/- input will probably work for the image editor stuff, but I'm also trying to figure out inputs for padding and other incremental # inputs. To make it even more complicated, some require a reset button. This might be better discussed in person during office hours or our next sync?

**Reply by serifluous** [reply]
argh, shouldn't have removed our sync for this week. I'll add one back for tomorrow—let's chat then.
